File: packages/api/aqapi/core/domain/value_object/validation_exception.py
from aqapi.core.messages.locale_string import LocaleString


class ValidationException(Exception):
    def __init__(self, error_type: str, message: LocaleString):
        super().__init__(message.en)
        self.error_type = error_type
        self.message = message

class ValueValidateException(ValidationException):
    """値オブジェクトのバリデーションエラーを表すカスタム例外クラス"""
    def __init__(self, value_name: LocaleString, error_type: str, message: LocaleString):
        super().__init__(error_type, message)
        self.value_name = value_name

class RelationValidateException(ValidationException):
    """関連バリデーションエラーを表すカスタム例外クラス"""
    def __init__(self, error_type: str, message: LocaleString):
        super().__init__(error_type, message)

File: packages/shared/core/validator/validationException.ts
import { LocaleString } from '../messages/localeString';

export class ValidationException extends Error {
  public readonly errorType: string;
  public readonly localeMessage: LocaleString;

  constructor(params: {
    errorType: string;
    message: LocaleString;
  }) {
    super(params.message.en);
    this.errorType = params.errorType;
    this.localeMessage = params.message;
    this.name = 'ValidationException';
  }
}

export class ValueValidateException extends ValidationException {
  public readonly valueName: LocaleString;

  /**
   * 値オブジェクトのバリデーションエラーを表すカスタム例外クラス
   */
  constructor(params: {
    valueName: LocaleString;
    errorType: string;
    message: LocaleString;
  }) {
    super({ errorType: params.errorType, message: params.message });
    this.valueName = params.valueName;
    this.name = 'ValueValidateException';
  }
}

export class RelationValidateException extends ValidationException {
  /**
   * 関連バリデーションエラーを表すカスタム例外クラス
   */
  constructor(params: {
    errorType: string;
    message: LocaleString;
  }) {
    super({ errorType: params.errorType, message: params.message });
    this.name = 'RelationValidateException';
  }
} 
